sdks/python/apache_beam/runners/dataflow/internal/apiclient.py

```diff
@@ -285,12 +285,26 @@ def shortstrings_registerer(encoding_name):
         indent=2, sort_keys=True)
 
   @staticmethod
+  def _build_default_job_name(user_name):
+    """Generates a default name for a job.
+
+    user_name is lowercased, and any characters outside of [-a-z0-9]
+    are removed. If necessary, the user_name is truncated to shorten
+    the job name to 63 characters."""
+    user_name = re.sub('[^-a-z0-9]', '', user_name.lower())
+    date_component = datetime.utcnow().strftime('%m%d%H%M%S-%f')
+    app_name = 'beamapp'
+    job_name = '{}-{}-{}'.format(app_name, user_name, date_component)
+    if len(job_name) > 63:
+      job_name = '{}-{}-{}'.format(app_name,
+                                   user_name[:-(len(job_name) - 63)],
```

Comment:
I did it that way because the **excess length** is `len(job_name) - 63`.
Otherwise, we would want to keep the first:
`63 - len(app_name) - len(date_component) - len('---')` (Because we add three dashes when we format)

I have switched it to truncate both appname and username together. What do you think?